### DISCUSSION TITLE
"batteries included" by default

### DISCUSSION THREAD

**Discussion by msteveb** [msteveb/jimtcl#228] [Ideas]
Currently ./configure takes a minimal approach that does not include things like math functions and utf8 by default. This can be a *gotcha* for new users. Advanced users can alway turn off features if necessary to reduce size, so I am thinking of making something like ./configure --full the default. Thoughts?

**Comment by gromgit** [comment]
Sorry for the long delay. I personally prefer the current "add what you need" instead of "remove what you don't", especially since a major selling point of Jim is _minimal_.

Have there been many instances of new users actually saying "I wish you'd made _full_ the default config?"

**Reply by msteveb** [reply]
Thanks, any feedback is useful. Yes, I've hand a number of users who were surprised that things like math functions weren't available. While developers familiar with jimtcl have a lot of control over what to include through configure settings, new users are much more likely to just do ./configure && make && make install. I want these users to have a good initial experience.

**Reply by gromgit** [reply]
Then by all means switch to "full" as default. I habitually run `./configure --help` on everything that has a configure script, so as long as it starts saying `--disable-utf8` etc., I'll adapt as necessary.

**Reply by frithrah** [reply]
The lack of maths functions by default did surprise me but it only took a minute to sort out. I agree that setting `--full` by default would make for a smoother introduction.

**Comment by msteveb** [comment]
I created https://github.com/msteveb/jimtcl/pull/235
